pyflakes/test/test_imports.py

```diff
@@ -1031,6 +1039,36 @@ def test_futureImportStar(self):
         from __future__ import *
         ''', m.FutureFeatureNotDefined)
 
+    def test_ignoresTypingImports(self):
+        """Ignores imports within 'if TYPE_CHECKING' checking normal code."""
+        self.flakes('''
+        from typing import TYPE_CHECKING
+        if TYPE_CHECKING:
+            from a import b
+        b()
+        ''', m.UndefinedName)
+
+    def test_ignoresTypingClassDefinition(self):
+        """Ignores definitions within 'if TYPE_CHECKING' checking normal code."""
+        self.flakes('''
+        from typing import TYPE_CHECKING
+        if TYPE_CHECKING:
+            class T:
+                pass
+        t = T()
+        ''', m.UndefinedName)
+
+    @skipIf(version_info < (3,), 'has type annotations')
```

Comment:
these can use `# type: () -> b` comments such that this is validated in python2 and python3

Reply:
I can't get this to work. I add an `assert False` right after the `continue` in `_collect_type_comments` and it never fires in the test (after I change to a ` # type: () -> b` comment)

Reply:
Oh! It's the `test_doctests` that runs the tests again with some extra stuff that breaks type comments. :-(

At least I can fix it now